src/basic/path-util.c

```diff
@@ -853,3 +854,28 @@ char *prefix_root(const char *root, const char *path) {
         strcpy(p, path);
         return n;
 }
+
+bool path_prefix_in_list(const char *path, const char *x, ...) {
+        const char *p;
+        va_list ap;
+
+        if (!path || !x)
+                return false;
+
+        va_start(ap, x);
+
+        if (path_startswith(path, x)) {
+                va_end(ap);
+                return true;
+        }
+
+        while ((p = va_arg(ap, const char*)))
+                if (path_startswith(path, p)) {
+                        va_end(ap);
+                        return true;
+                }
+
+        va_end(ap);
+
+        return false;
+}
```

Comment:
Idea was that if you want to call with empty list then you have to explicitly specify NULL as second argument. I think we do the same for some of our public APIs, for example,

```C
int sd_bus_emit_interfaces_added(sd_bus *bus, const char *path, const char *interface, ...) _sd_sentinel_
```
I figured that doing the same would be probably OK.